volatility/framework/plugins/linux/check_idt.py

```diff
@@ -0,0 +1,94 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import List
+
+from volatility.framework import interfaces, renderers, constants, contexts, exceptions, symbols
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.symbols import linux
+from volatility.plugins.linux import lsmod
+
+vollog = logging.getLogger(__name__)
+
+
+class Check_idt(interfaces.plugins.PluginInterface):
+    """ Checks if the IDT has been altered """
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.TranslationLayerRequirement(name='primary',
+                                                     description='Memory layer for the kernel',
+                                                     architectures=["Intel32", "Intel64"]),
+
+            requirements.SymbolTableRequirement(name="vmlinux", description="Linux kernel symbols"),
+            
+            requirements.PluginRequirement(name='lsmod', plugin=lsmod.Lsmod, version=(1, 0, 0))
+        ]
+
+    def _generator(self):
+        vmlinux = contexts.Module(self.context, self.config['vmlinux'], self.config['primary'], 0)
+
+        modules = lsmod.Lsmod.list_modules(self.context, self.config['primary'], self.config['vmlinux'])
+
+        handlers = linux.LinuxUtilities.generate_kernel_handler_info(self.context, self.config['primary'], self.config['vmlinux'], modules)
+
+
+        is_32bit = not symbols.symbol_table_is_64bit(self.context, self.config["vmlinux"])
+
+        tblsz = 256
+
+        # hw handlers + system call
+        check_idxs = list(range(0, 20)) + [128]
+
+        if is_32bit:
+            if vmlinux.has_type("gate_struct"):
+                idt_type = "gate_struct"
+            else:
+                idt_type = "desc_struct"
+        else:
+            if vmlinux.has_type("gate_struct64"):
+                idt_type = "gate_struct64"
+            elif vmlinux.has_type("gate_struct"):
+                idt_type = "gate_struct"
+            else:
+                idt_type = "idt_desc"
+
+        # this is written as a list b/c there are supposdly kernels with per-CPU IDTs
```

Comment:
Let me know what you want me to do for this

Reply:
The bit above looks fine, if there's a few different cases where it can differ, then we need to differentiate them.  I dunno if it would be more confusing to check whether `gate_struct` is there first (since it's in both 32 and 64 bit ones), but it looks like `gate_struct64` may need to take precedence so best keep it this way.

If it's the bit below, then I'd need to go check the vol2 code, but even if there were multiple tables, you'll only get a single object back from `object_from_symbol`, so even if it's an array, you'd still be cycling through a list of one each time...

Reply:
From the vol2 code, it looks like this is priming the code to end up dealing with `idt_table` were it ever to be a list, but I think the structure name would change in that case?  It's all internal (ie, not an exposed API), so changing it later shouldn't have any effect, so I'd say drop the list here and if the structure does ever change for some reason, we can put the code back in then.

Reply:
sounds good